Compilation for ARM target fails because there is still a non-opaque pointer API in use

**Describe the bug**

Compilation for ARM target fails because there is still a non-opaque pointer API in use.
```
/storage/rkeryell/SYCL/sycl-mlir/llvm/llvm/lib/Target/ARM/ARMFastISel.cpp:2183:22:
error: ‘getInt32PtrTy’ is not a member of ‘llvm::Type’
 2183 |   Type *GVTy = Type::getInt32PtrTy(*Context, /*AS=*/0);
      |                      ^~~~~~~~~~~~~
/storage/rkeryell/SYCL/sycl-mlir/llvm/llvm/lib/Target/ARM/ARMFastISel.cpp: In member function ‘unsigned int {anonymous}::ARMFastISel::ARMLowerPICELF(const llvm::GlobalValue*, llvm::MVT)’:
/storage/rkeryell/SYCL/sycl-mlir/llvm/llvm/lib/Target/ARM/ARMFastISel.cpp:2967:50: error: ‘getInt32PtrTy’ is not a member of ‘llvm::Type’
 2967 |       MF->getDataLayout().getPrefTypeAlign(Type::getInt32PtrTy(*Context));
      |                                                  ^~~~~~~~~~~~~
```

It looks like this is because the following commit is missing from upstream:
```
commit fd05c34b18fbc3f37494bef9118c368f0c11595a
Author: Bjorn Pettersson <...@ericsson.com>
Date:   Mon Jul 31 16:32:44 2023 +0200

    Stop using legacy helpers indicating typed pointer types. NFC
    
    Since we no longer support typed LLVM IR pointer types, the code can
    be simplified into for example using PointerType::get directly instead
    of using Type::getInt8PtrTy and Type::getInt32PtrTy etc.
    
    Differential Revision: https://reviews.llvm.org/D156733
```
It would be nice to check that most of the current commits are actually in the `sycl` branch.

**To Reproduce**

I am actually configuring and compiling the `sycl-mlir` branch with
` -DLLVM_TARGETS_TO_BUILD="X86;NVPTX;ARM;AArch64"`
but the commit is also missing from the `sycl` branch.

**Environment (please complete the following information):**

- OS: Ubuntu 22.04
- Target device and vendor: ARM
- DPC++ version: ff61613e225a from today.

--- a/llvm/lib/Target/ARM/ARMFastISel.cpp
+++ b/llvm/lib/Target/ARM/ARMFastISel.cpp
@@ -2180,7 +2180,7 @@ unsigned ARMFastISel::ARMSelectCallOp(bool UseReg) {
 
 unsigned ARMFastISel::getLibcallReg(const Twine &Name) {
   // Manually compute the global's type to avoid building it when unnecessary.
-  Type *GVTy = Type::getInt32PtrTy(*Context, /*AS=*/0);
+  Type *GVTy = PointerType::get(*Context, /*AS=*/0);
   EVT LCREVT = TLI.getValueType(DL, GVTy);
   if (!LCREVT.isSimple()) return 0;
 
@@ -2964,7 +2964,7 @@ unsigned ARMFastISel::ARMLowerPICELF(const GlobalValue *GV, MVT VT) {
       /*AddCurrentAddress=*/UseGOT_PREL);
 
   Align ConstAlign =
-      MF->getDataLayout().getPrefTypeAlign(Type::getInt32PtrTy(*Context));
+      MF->getDataLayout().getPrefTypeAlign(PointerType::get(*Context, 0));
   unsigned Idx = MF->getConstantPool()->getConstantPoolIndex(CPV, ConstAlign);
   MachineMemOperand *CPMMO =
       MF->getMachineMemOperand(MachinePointerInfo::getConstantPool(*MF),

